lib/queuePopulator/IngestionPopulator.js

```diff
@@ -749,29 +766,55 @@ class IngestionPopulator {
         return false;
     }
 
+    _processLogReaderEntries(logReader, params, cb) {
+        // skipping if circuit breaker tripped
+        if (this._circuitBreaker.state !== BreakerState.Nominal) {
+            this.log.debug('circuitbreaker tripped, retrying later', {
+                zenkoBucket: logReader.getTargetZenkoBucketName(),
+                location: logReader.getLocationConstraint(),
+            });
+            return cb();
+        }
+        // skipping if previous batch still in progress
+        if (logReader.isBatchInProgress()) {
+            this.log.debug('ingestion batch still in progress, skipping', {
+                zenkoBucket: logReader.getTargetZenkoBucketName(),
+                location: logReader.getLocationConstraint(),
+            });
+            return cb();
+        }
+        // skipping if location is paused
+        if (this._pausedLocations[logReader.getLocationConstraint()] !== undefined) {
+            this.log.debug('location paused, skipping', {
+                zenkoBucket: logReader.getTargetZenkoBucketName(),
+                location: logReader.getLocationConstraint(),
+            });
+            return cb();
+        }
```

Comment:
does not seem related : did we not already stop ingestion when location was paused?
or is it juste an "optimization", to skip the last few entries which we may still get immediately when ingestion is paused

Reply:
actually, this one is associated with another ticket (BB-399) and is the latest one : should we move it a separate, stacked PR maybe?

Reply:
yes i forgot to target the BB-399 PR, fixed that